R/vars_manager.R

```diff
@@ -44,93 +44,112 @@ make_multi_dict <- function(
+        if (is.data.frame(filters) && length(filters) > 0) {
+            col_names <- colnames(filters)
+            if (
+                ("col" %in% col_names) &&
+                ("search" %in% col_names) &&
+                ("min" %in% col_names) &&
+                ("max" %in% col_names)
+            ) {
+                row1 <- filters[1,]
+                col_name <- row1$col[1]
+
+                if (is.character(col_name) && tolower(col_name) == "value") {
+                    search <- row1$search[1]
+                    min <- row1$min[1]
+                    max <- row1$max[1]
+                    obj_numeric <- is.numeric(obj)
+                    if (is.character(search)) {
+                        obj_filtered <- obj_filtered[grepl(search, obj_filtered, ignore.case=TRUE)]
+                    }
+                    if (is.numeric(min) && obj_numeric) {
+                        obj_filtered <- obj_filtered[obj_filtered >= min]
+                    }
+                    if (is.numeric(max) && obj_numeric) {
+                        obj_filtered <- obj_filtered[obj_filtered <= max]
+                    }
                 }
             }
         }
-    }
 
-    abbrev <- !is.null(abbrev_len) && length(obj_filtered) > abbrev_len
+        start <- `if`(!is.null(page_size), page_size * page + 1, 1)
+        has_next_page <- `if`(!is.null(page_size), length(obj_filtered) > (start + page_size - 1), FALSE)
+        end <- `if`(has_next_page, start + page_size - 1, length(obj_filtered))
 
-    if (
-        (is.character(sort_by) && length(sort_by) == 1 && tolower(sort_by) == "value")
-    ) {
-        if (is.logical(ascending)) {
-            decreasing <- !ascending
+        if (
+            (is.character(sort_by) && length(sort_by) == 1 && tolower(sort_by) == "value")
+        ) {
+            if (is.logical(ascending)) {
+                decreasing <- !ascending
+            } else {
+                decreasing <- FALSE
+            }
+
+            if (is.character(obj_filtered)) {
+                obj_sorted <- obj_filtered[order(tolower(obj_filtered), decreasing=decreasing, na.last=decreasing)]
+            } else {
+                obj_sorted <- obj_filtered[order(obj_filtered, decreasing=decreasing, na.last=decreasing)]
+            }
+            
+            obj_pre <- obj_sorted[start:end]
         } else {
-            decreasing <- FALSE
+            obj_pre <- obj_filtered[start:end]
         }
 
-        if (is.character(obj_filtered)) {
-            obj_sorted <- obj_filtered[order(tolower(obj_filtered), decreasing=decreasing, na.last=decreasing)]
-        } else {
-            obj_sorted <- obj_filtered[order(obj_filtered, decreasing=decreasing, na.last=decreasing)]
+        data <- list()
+        row_names <- list()
+
+        for (i in 1:length(obj_pre)) {
+            data[[i]] = list(as.character(obj_pre[i]))
+            if (is.null(names(obj_pre))) {
+                row_names[i] <- as.character(i)
```

Comment:
I don't think this is quite right, `row_names` should match the page you're delivering (`start` to `end`)